File: ckgame-java/src/main/java/com/ckgame/world/buildings/CountyBuilding.java
package com.ckgame.world.buildings;

/** 省份中的一座建筑及其等级。 */
public final class CountyBuilding {
    public BuildingKind kind;
    public int level;

    public CountyBuilding(BuildingKind kind, int level) {
        this.kind = kind;
        this.level = level;
    }

    public CountyBuilding(BuildingKind kind) {
        this(kind, 0);
    }

    public boolean canUpgrade() {
        return level < kind.maxLevel();
    }

    public int upgradeCost() {
        return kind.upgradeCost(level);
    }
}

File: ckgame-ts/src/world/buildings/CountyBuilding.ts
import { BuildingKind, buildingMaxLevel, buildingUpgradeCost } from './BuildingKind.js';

/** 省份中的一座建筑及其等级。 */
export class CountyBuilding {
  kind: BuildingKind;
  level: number;

  constructor(kind: BuildingKind, level: number = 0) {
    this.kind = kind;
    this.level = level;
  }

  canUpgrade(): boolean {
    return this.level < buildingMaxLevel(this.kind);
  }

  upgradeCost(): number {
    return buildingUpgradeCost(this.kind, this.level);
  }
}
